### DISCUSSION TITLE
Error when using Server Action in Client Component

### DISCUSSION THREAD

**Discussion by nv-cho** [vercel/next.js#55694] [App Router]
What I'm trying to do is the following: use a client component to submit a form and then write to a Google Sheet with the information received using a server action.

* Client component:

<img width="564" alt="image" src="https://github.com/vercel/next.js/assets/86693266/fdcca94f-bd8f-4154-9946-ad7044e25c88">

* sendEmail (server action)

<img width="520" alt="image" src="https://github.com/vercel/next.js/assets/86693266/9c9339dc-f069-454c-90b6-25fbd9fec338">

* googlesheet functions (server action)

<img width="564" alt="image" src="https://github.com/vercel/next.js/assets/86693266/44ec4a3f-71e9-4a91-a311-c5257497213c">

My problem here is that after submit the form I receive this error:

<img width="980" alt="image" src="https://github.com/vercel/next.js/assets/86693266/875914fe-99df-4a01-a397-3a0ae9c19142">

IMPORTANT: The functions actually work, as my spreadsheet is written with each submission

Thanks! 🫶

**Comment by icyJoseph** [comment]
I couldn't really reproduce this. What Next.js version do you have in that project?

**Reply by nv-cho** [reply]
13.5.2

**Reply by icyJoseph** [reply]
Are you perhaps passing something to `Form` from a Server Component to a Client Component?

**Reply by nv-cho** [reply]
The form is triggering sendEmail which is a Server Action, but that's everything :/ I'm switching to API routes since I couldn't find a solution

**Reply by icyJoseph** [reply]
Do you have a link to the repository? I feel like something is missing here :/

**Reply by nv-cho** [reply]
It's a single page landing, I cannot share the repo because it's from my work but there isn't much more. A homepage with this form inside and that's it
